File: output/deepseek_chat/avatar/direct_translation/python/java/atcoder_ARC086_B.java

import java.util.*;
import java.io.*;

public class atcoder_ARC086_B {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int N = Integer.parseInt(br.readLine());
        int[] a = Arrays.stream(br.readLine().split(" ")).mapToInt(Integer::parseInt).toArray();
        int maxVal = 0;
        int maxSign = 0;
        int maxIndex = -1;
        for (int i = 0; i < a.length; i++) {
            if (Math.abs(a[i]) > maxVal) {
                maxVal = Math.abs(a[i]);
                maxSign = sign(a[i]);
                maxIndex = i;
            }
        }
        if (maxSign == 0) {
            System.out.println(0);
        } else if (maxSign == 1) {
            System.out.println(2 * N - 1);
            for (int i = 0; i < N; i++) {
                System.out.println((maxIndex + 1) + " " + (i + 1));
            }
            for (int i = 0; i < N - 1; i++) {
                System.out.println((i + 1) + " " + (i + 2));
            }
        } else {
            System.out.println(2 * N - 1);
            for (int i = 0; i < N; i++) {
                System.out.println((maxIndex + 1) + " " + (i + 1));
            }
            for (int i = N - 1; i > 0; i--) {
                System.out.println((i + 1) + " " + i);
            }
        }
    }

    private static int sign(int x) {
        return (x > 0) ? 1 : (x < 0) ? -1 : 0;
    }
}

File: output/gpt_4o/avatar/direct_translation/java/python/failed_translation/atcoder_ARC086_B.py

n = int(input())
a = list(map(int, input().split()))
r = []
m = 0
for i in range(1, n):
    if abs(a[m]) < abs(a[i]):
        m = i
for i in range(n):
    if (a[m] >= 0) ^ (a[i] >= 0):
        r.append(f"{m + 1} {i + 1}")
if a[m] >= 0:
    for i in range(1, n):
        r.append(f"{i} {i + 1}")
else:
    for i in range(n, 1, -1):
        r.append(f"{i} {i - 1}")
print(len(r))
for x in r:
    print(x)

